firefox-ios/Client/Frontend/UserContent/UserScripts/MainFrame/AtDocumentEnd/MetadataHelper.js

```diff
@@ -9,7 +9,7 @@ const metadataParser = require("page-metadata-parser/parser.js");
 
 function MetadataWrapper() {
   this.getMetadata = function() {
-    let metadata = metadataParser.getMetadata(document, document.URL);
+    let metadata = metadataParser.getMetadata(document, location.origin);
```

Comment:
Note: this change was suggested by Issam (ty), and is part of the fix for the incorrect favicon URLs, which are actually coming from two different places, one of which being the JS metadata parser.